doc/content/hpc_build_tips.md

```diff
@@ -0,0 +1,156 @@
+HOME_DIRECTORY_SYM_LINK=$(realpath -P $DIRECTORY_WHERE_YOU_HAVE_CARDINAL)
+export NEKRS_HOME=$HOME_DIRECTORY_SYM_LINK/cardinal/install
+
+# These environment variables indicate how many processes you want to build with. Be careful with the number
+# of jobs specified, most HPC systems limit the number of processes you can run with on login nodes.
+export JOBS=8
+export LIBMESH_JOBS=8
+export MOOSE_JOBS=8
+```
+
+The best place to put these exports is your `.bashrc` to avoid needing to export these environment variables every time you spin up a new shell. Once these variables have been exported and you've loaded a set of modules that you believe will compile Cardinal, you can run MOOSE diagnostics to make sure you haven't missed anything:
+
+```bash
+./contrib/moose/scripts/diagnostics.sh
+```
+
+If critical errors are reported at this stage, it is likely because your current module set does not meet MOOSE's requirements. If you receive an error stating that `jinja2` is missing, you can safely move on.
+
+**Note:** If you want to build NekRS with GPU support, you need GPU-enabled compilers (CUDA compilers, OpenCL compilers, etc.) on the login node. Some HPC systems only allow users to load those modules on nodes which contain GPUs. If that is the case for you, you'll need to build with a job script (see the next session). To enable a GPU build, set one of the following environment variables in Cardinal's makefile to `1` (depending on the specific GPUs on your system): `OCCA_CUDA_ENABLED`, `OCCA_HIP_ENABLED`, or `OCCA_OPENCL_ENABLED`. You'll also need to make sure you load modules with GPU compilers.
+
+From here, you can run the commands below to build MOOSE's submodules. We recommend building with `nohup` or `screen` to avoid getting timed out while you're connected to a login node.
+
+```
+nohup ./contrib/moose/scripts/update_and_rebuild_petsc.sh &
+```
+
+```
+nohup ./contrib/moose/scripts/update_and_rebuild_libmesh.sh &
+```
+
+```
+nohup ./contrib/moose/scripts/update_and_rebuild_wasp.sh &
+```
+
+Building libMesh may fail due to a lack of libraries that are normally included by default on Linux machines. The most common culprit is `libxdr`, which is required for XDR binary file output in libMesh/MOOSE. The vast majority of the time you'll use Exodus binary output when running Cardinal, and so XDR can be disabled by adding two flags to `update_and_rebuild_libmesh.sh`.
+
+```
+./contrib/moose/scripts/update_and_rebuild_libmesh.sh --disable-xdr-required --disable-xdr
+```
+
+If you didn't get any build errors when building PETSc/libMesh/WASP, you can run
+
+```
+nohup make -j8 MAKEFLAGS=-j8 &
+```
+
+to build MOOSE, Cardinal, and all of Cardinal's dependencies. Occasionally the MOOSE Solid Mechanics module will fail to build due to missing `F77` compilers. If that is the case, you can either find mpif77 compilers in a different module set or disable the Solid Mechanics module by setting `SOLID_MECHANICS := no` in Cardinal's makefile.
+
```

Comment:
```suggestion
The only MOOSE module you absolutely must have enabled is the `REACTOR` module (some of Cardinal's source files link against utilities in that module). Otherwise, feel free to disable all others.
```